### PR TITLE
Option to fall back to Piped proxy when Piped proxy disabled and video unreachable

### LINKED ISSUE #4038: Fallback to Piped Proxy Option
### Describe your suggested feature

An option under Audio and Video > Advanced that ties in with the "Disable Piped proxy" option. When "Fallback to Piped proxy" is enabled, LibreTube will first attempt direct connection to YouTube. Then if the playback fails, attempt connecting using the selected Piped instance.

This should allow YouTube Music content to load successfully without needing to manually switch back on the Piped proxy and would provide a good mix of speed to load content and less videos that fail to play back.

### Other details

The option can be greyed out if "Disable Piped proxy" is not enabled and checked off by default after enabling "Disable Piped proxy"

### Acknowledgements

- [X] I have searched the existing issues and this is a new ticket, **NOT** a duplicate or related to another open issue.
- [X] I have written a short but informative title.
- [X] I will fill out all of the requested information in this form.

### PR DESCRIPTION
**Pull request by Bnyro**
closes  #4038

### CODE REVIEW

File: app/src/main/java/com/github/libretube/helpers/ProxyHelper.kt [hunk 1]
```diff
@@ -10,6 +10,8 @@ import okhttp3.OkHttpClient
 import okhttp3.Request
 
 object ProxyHelper {
+    private val client by lazy { OkHttpClient.Builder().build() }
```
Comment (FireMasterK) [inline app/src/main/java/com/github/libretube/helpers/ProxyHelper.kt:13]: Why not use the cronet client here? :thinking: 
We have an adaptor for Cronet's engine for OkHttp

**Comment by Bnyro** [comment]
Is there a video to test this with? For me it appears that YT music videos work as expected even without the Proxy somehow?

**Comment by ehea617** [comment]
I'm currently getting an error at the bottom of the screen when opening any song in the "YT Music Songs" search filter, "Initial player response is not valid"

This error message appears regardless of the song selected or combination of options of piped proxy enabled/disabled and fallback enabled/disabled.

**Comment by Bnyro** [comment]
Try a different instance, see https://github.com/TeamPiped/Piped/issues/2487

**Comment by Bnyro** [comment]
That's not related to the changes in this PR.

**Comment by ehea617** [comment]
Thanks for the info - I can confirm this works great after I switched the instance over to projectsegfau.lt and do not notice any issues specific to this now.

**Comment by Bnyro** [comment]
Alright, thanks for testing!

**Pull request merged**